shapepipe/modules/merge_headers_package/merge_headers.py

```diff
@@ -42,24 +48,103 @@ def merge_headers(input_file_list, output_dir):
+        else:
+            file_path = datadir+pattern+"key"+ext
+            final_file[key] = np.load(file_path, allow_pickle=True)
+    # Close file
+    final_file.commit()
+    final_file.close()
+
+def unique_exposure_list(input_file_list, header_dir, file_pattern, ext, ext_header):
+    """unique_exposure_list.
+
+    Extract unique exposure ids from file list.
+
+    Parameters
+    ----------
+    input_file_list : list
+        list of input files
+    header_dir : str
+        directory of headers
+    file_pattern : str
+        text in filename 
+    ext : str
+        file extension, eg .npy, .head
+    ext_header : bool
+        Uses external scamp header if ``True`` 
 
-    for file_path in input_file_list:
-        # Extract file path
-        file_path_scalar = file_path[0]
-        # Check file name against pattern
+    Returns
+    -------
+    list
+        List of unique exposure numbers
+    """
+    file_list = []
+    if ext_header:
+        ext = '\-\d{1,2}' + ext 
+    # extract keys
+    for file in input_file_list:
+        file_path_scalar = file[0]
         file_name = os.path.split(file_path_scalar)[1]
-        file_base_name = os.path.splitext(file_name)[0]
-        pattern_split = re.split(pattern, file_base_name)
+        root = re.split(ext, file_name)[0]
+        pattern_split = re.split(file_pattern, root)
         if len(pattern_split) < 2:
             raise IndexError(
                 f'Regex "{pattern}" not found in base name "{file_base_name}".'
             )
         key = pattern_split[1]
-        # Load Numpy binary file
-        final_file[key] = np.load(file_path_scalar, allow_pickle=True)
+        
+        file_list.append(key)
+    # only keep unique keys
+    unique_keys = list(set(file_list))
 
-    # Close file
-    final_file.commit()
-    final_file.close()
+    return unique_keys
+
+def create_joint_header(n_hdu, key, header_dir, pattern, ext):
+    """create_joint_header.
+
+    Packages scamp headers for ccds into a numpy array per exposure.
+     
+    Parameters
+    ----------
+    n_hdu : int
+        total number of ccds
+    key : str
+        exposure number
+    header_dir : str
+        directory where headers are stored
+    pattern : str
+        file pattern, e.g. headers-
+    ext : str
+        header extension 
+    Returns
+    -------
+    list
+        compilation of headers corresponding to exposure number
+    """
+
+    header_file = np.zeros(n_hdu, dtype='O')
+    for idx in range(1, n_hdu + 1):
+        filepath= header_dir+'/'+pattern+key+'-'+str(idx)+ext
+        print(filepath)
+        # check for file                                                
+        if os.path.exists(filepath):
+            h=Header.fromfile(filepath,sep='\n',endcard=False,padding=False)
+            try:
+                w = WCS(h)
+            except Exception:
+                print(f'WCS error for file {exp_path}')
+                raise
```

Comment:
After raising the exepction I think the program will end, so the code after that will not be run?